src/transformers/models/bert/modeling_bert.py

```diff
@@ -992,7 +1146,15 @@ def forward(
             encoder_hidden_shape = (encoder_batch_size, encoder_sequence_length)
             if encoder_attention_mask is None:
                 encoder_attention_mask = torch.ones(encoder_hidden_shape, device=device)
-            encoder_extended_attention_mask = self.invert_attention_mask(encoder_attention_mask)
+
+            if use_sdpa_attention_masks:
+                # Expand the attention mask for SDPA.
+                # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
```

Comment:
```suggestion
                # [bsz, seq_len] -> [bsz, 1, src_seq_len, src_seq_len]
```
the `_prepare_4d_attention_mask_for_sdpa ` (and the expand) is not capable of that, it's always expanding to a squared attention mask

Reply:
It is not true, `_prepare_4d_attention_mask_for_sdpa` has a `tgt_len` argument.

Reply:
you cannot expand in two direction 😉

Reply:
I've updated the comment to use only seq_len